packages/google_maps_flutter/google_maps_flutter/ios/Classes/GoogleMapHeatmapController.h

```diff
@@ -0,0 +1,34 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#import <Flutter/Flutter.h>
+#import <GoogleMaps/GoogleMaps.h>
+@import GoogleMapsUtils;
+
+// Defines heatmap UI options writable from Flutter.
+@protocol FLTGoogleMapHeatmapOptionsSink
+- (void)setWeightedData:(NSArray<GMUWeightedLatLng *> *)weightedData;
+- (void)setGradient:(GMUGradient *)gradient;
+- (void)setOpacity:(double)opacity;
+- (void)setRadius:(int)radius;
+- (void)setMap;
+@end
+
+// Defines heatmap controllable by Flutter.
+@interface FLTGoogleMapHeatmapController : NSObject <FLTGoogleMapHeatmapOptionsSink>
+@property(atomic, readonly) NSString *heatmapId;
```

Comment:
can you explain why heatmapId needs to be `atomic`?

Reply:
I saw that we did have some other (very old) controllers having this property to be atomic. They were mostly introduced in the first initial GoogleMap patch: https://github.com/flutter/plugins/commit/2db1f4b4f95faf6d307514842d567b96bb360dc2
I don't think they need to be atomic, these values were never accessed from different threads, and making them atomic introduces unnecessary latencies. The other controllers are out of scope of this PR, so let's not worry about them in this PR. 

I will create a separate PR to make all those properties nonatomic.

CC @stuartmorgan

Reply:
Done